src/pages/MainPage.vue

```diff
@@ -179,22 +143,13 @@
         >
           <road
             :selected-subjects="roads[roadId].contents.selectedSubjects"
-            :subjects="subjectsInfo"
             :road-i-d="roadId"
             :current-semester="currentSemester"
             :adding-from-card="addingFromCard && activeRoad===roadId"
             :item-adding="itemAdding"
-            :subjects-index="subjectsIndexDict"
-            :generic-courses="genericCourses"
-            :generic-index="genericIndexDict"
             :drag-semester-num="(activeRoad===roadId) ? dragSemesterNum : -1"
             @add-at-placeholder="addAtPlaceholder"
```

Comment:
Moving this function into Class.vue would save a lot of $emits.

Reply:
Wait, what function are you referring to here?

Reply:
Just kidding, I see what you're referring to now (I'm quite tired). Will go ahead and move that.